task-sdk/tests/task_sdk/execution_time/test_context.py

```diff
@@ -542,8 +544,12 @@ def test_getitem_uri_ref(
     def test_source_task_instance_xcom_pull(self, mock_supervisor_comms, accessor):
         events = accessor[Asset("2")]
         assert len(events) == 1
+
+        mock_dag_run = mock.Mock(dag_id="d1", run_id="r1")
+        mock_supervisor_comms.send.side_effect = [mock_dag_run]
         source = events[0].source_task_instance
-        assert source == AssetEventSourceTaskInstance(dag_id="d1", task_id="t2", run_id="r1", map_index=-1)
+        assert source == AssetEventSourceTaskInstance(dag_run=mock_dag_run, task_id="t2", map_index=-1)
+        mock_supervisor_comms.send.assert_called_once_with(GetDagRun(dag_id="d1", run_id="r1"))
```

Comment:
```suggestion
        assert mock_supervisor_comms.calls = [
            call(GetDagRun(dag_id="d1", run_id="r1")),
        ]
```

Should we go with what you suggested in the dev list a few months ago?